AcsfToolsCommands.php

```diff
@@ -206,72 +181,260 @@ public function ml($cmd, $command_args = '', $command_options = '', $options = [
+   * @bootstrap site
+   * @params $cmd
+   *   The drush command you want to run against all sites in your factory.
+   * @params $command_args Optional.
+   *   A quoted, space delimited set of arguments to pass to your drush command.
+   * @params $command_options Optional.
+   *   A quoted space delimited set of options to pass to your drush command.
+   * @option domain-pattern
+   *   Pattern / keyword to check for choosing the domain for uri parameter.
+   * @option profiles
+   *   Target sites with specific profiles. Comma list.
+   * @option use-https
+   *   Use secure urls for drush commands.
+   * @usage drush acsf-tools-mlc st
+   *   Get output of `drush status` for all the sites.
+   * @usage drush acsf-tools-mlc cget "'system.site' 'mail'"
+   *   Get value of site_mail variable for all the sites.
+   * @usage drush acsf-tools-mlc upwd "'admin' 'password'"
+   *   Update user password.
+   * @usage drush acsf-tools-mlc cget "'system.site' 'mail'" "'format=json' 'interactive-mode'"
+   *   Fetch config value in JSON format.
+   * @usage drush acsf-tools-mlc cr --delay=10
+   *   Run cache clear on all sites with delay of 10 seconds between each site.
+   * @usage drush acsf-tools-mlc cron --use-https=1
+   *   Run cron on all sites using secure url for URI.
+   * @usage drush acsf-tools-mlc cron --pattern=collection
+   *   Run cron on all sites using domain that contains the pattern "collection" for URI.
+   *   By default it uses first custom domain. If no domain available it uses acsitefactory.com domain.
+   *   From abc.collection.xyz.com and abc.xyz.acsitefactory.com it will choose abc.collection.xyz.com domain.
+   * @aliases sfmlc,acsf-tools-mlc
+   */
+  public function mlc($cmd, $command_args = '', $command_options = '', $options = ['domain-pattern' => '', 'profiles' => '', 'use-https' => 0]) {
+    // Look for list of sites and loop over it.
+    $sites = $this->getSites();
+    if (empty($sites)) {
+      return;
+    }
+
+    $command_args = $this->getCommandArgs($command_args);
+
+    // Parse list of options to be passed to the drush sub-command being invoked.
+    $drush_command_options = $this->getCommandOptions($command_options);
+
+    // Command always passes the default option as `yes` irrespective if `--no`
+    // option used. Pass confirmation as `no` if use that.
+    if ($options['no']) {
+      $drush_command_options['no'] = TRUE;
+    }
+
+    $profiles = [];
+    if (!empty($options['profiles'])) {
+      $profiles = explode(',', $options['profiles']);
+      unset($options['profiles']);
+    }
+
+    $processes = [];
+
+    foreach ($sites as $delta => $details) {
+      $domain = $this->getDomain($details, $options);
+      $process = $this->prepareCommand($domain, $details, $cmd, $command_args, $drush_command_options, $profiles);
+      if (empty($process)) {
+        continue;
+      }
+
+      $processes[$domain] = $process;
+      $this->output()->writeln("\n=> Executing command on $domain");
```

Comment:
I guess the screen output becomes awful when we run on multiple ones at the same time

Reply:
not really, output is all done in the loop below after execution is finished for particular domain
https://github.com/acquia/acsf-tools/pull/108/files#diff-16128690f3f86ae811dda9f676489477R298

Reply:
you may have a buffer size issue then

Reply:
we will have something like
executing on abc
executing on xyz

The command executed successfully for the site abc
OUTPUT
The command executed successfully for the site xyz
OUTPUT

Reply:
about buffer size, any idea or link detail the issue or a way for me to reproduce?

Reply:
I am surprised that we are doing that today but I may have missed something.
What I am worried about is getting a huge output from one command that would end up reaching a memory limit or any other limitation.
But if you are sure we do that today already we should be fine.

Reply:
yep, we do that already today